src/components/Dialog/DialogOverlap.tsx

```diff
@@ -1,17 +1,35 @@
-import React, { FC, PropsWithChildren, ReactNode, useEffect, useState } from 'react'
+import React, { FC, PropsWithChildren, ReactNode, useEffect, useMemo, useState } from 'react'
 import { CSSTransition } from 'react-transition-group'
-import styled from 'styled-components'
-
-import { Theme, useTheme } from '../../hooks/useTheme'
+import { tv } from 'tailwind-variants'
 
 type Props = PropsWithChildren<{
   isOpen: boolean
 }>
 
-const transitionClassName = 'shr-dialog-transition'
+const dialogOverlap = tv({
+  base: [
+    'shr-absolute',
+    'shr-z-overlap-base',
+    '[&.shr-dialog-transition-appear]:shr-opacity-0',
+    '[&.shr-dialog-transition-appear-active]:shr-transition-opacity',
+    '[&.shr-dialog-transition-appear-active]:shr-duration-500',
+    '[&.shr-dialog-transition-appear-active]:shr-ease-in-out',
+    '[&.shr-dialog-transition-appear-active]:shr-opacity-100',
+    '[&.shr-dialog-transition-enter]:shr-opacity-0',
+    '[&.shr-dialog-transition-enter-active]:shr-transition-opacity',
+    '[&.shr-dialog-transition-enter-active]:shr-duration-300',
+    '[&.shr-dialog-transition-enter-active]:shr-ease-in-out',
+    '[&.shr-dialog-transition-enter-active]:shr-opacity-100',
+    '[&.shr-dialog-transition-exit]:shr-opacity-100',
+    '[&.shr-dialog-transition-exit-active]:shr-transition-opacity',
+    '[&.shr-dialog-transition-exit-active]:shr-duration-300',
+    '[&.shr-dialog-transition-exit-active]:shr-ease-in-out',
+    '[&.shr-dialog-transition-exit-active]:shr-opacity-0',
+  ],
+})
```

Comment:
あまり細かく見れていないのですが、isOpenを渡して、単純にclassNameを置き換えるとかでは微妙ですかね？
`react-transition-group`が不要になるんじゃないかな〜とか思っております。
```suggestion
const dialogOverlap = tv({
  base: ['shr-absolute', 'shr-z-overlap-base', 'shr-transition-opacity', 'shr-ease-in-out'],
  variants: {
    isOpen: {
      false: 'shr-opacity-0 shr-duration-500',
      true: 'shr-opacity-100 shr-duration-300',
    },
  },
})
```

Reply:
DOM 上に存在する要素であれば variants でも問題なさそうでしたが、isOpen に依って DOM 自体を出し分けているため駄目そうでした。

Reply:
ちなみにこのコミットで解決しました!
`-active` ではなく `-done` を使ったらうまくいきました。
https://github.com/kufu/smarthr-ui/pull/3947/commits/cdcbe305e9b8e675f1db4f45d47c5d111d404327

Reply:
> isOpen に依って DOM 自体を出し分けているため駄目そうでした。

なるほどです〜！ご確認ありがとうございます:pray: